linux/deploy_vm/deploy_vm_from_ova.yml

```diff
@@ -130,6 +131,7 @@
   loop_control:
     loop_var: vm_cdrom
   when:
+    - guest_os_ansible_distribution is defined
     - guest_os_ansible_distribution == "Ubuntu"
     - vm_existing_cdrom_list is defined
     - vm_existing_cdrom_list | length > 0
```

Comment:
I think you can directly put line 92 ~ line 137 into a block with when: ova_guest_os_type in ['photon', 'ubuntu', 'amazon', 'flatcar', 'rhcos']